node.go

```diff
@@ -87,12 +97,69 @@ func (node *emptyNode) isEmpty() bool {
 
 var _ ProofNode = (*node)(nil)
 
+func (node *node) GetHash() []byte {
+	parent, ok := node.parent.(*interiorNode)
+	if !ok {
+		return nil
+	}
+	switch parent.leftChild.(type) {
+	case *emptyNode:
+		if &parent.leftChild.(*emptyNode).node == node {
+			return parent.leftHash
+		}
+	case *userLeafNode:
+		if &parent.leftChild.(*userLeafNode).node == node {
+			return parent.leftHash
+		}
+	default:
+		return nil
+	}
+
+	return parent.rightHash
+}
+
+func (node *interiorNode) GetHash() []byte {
+	return node.hash()
```

Comment:
If ```interiorNode``` is implementing ```ProofNode```, I think it should have the functionality of ```node```'s ```GetHash```, no? After all, the ```interiorNode```s in the proof should only expose the hash of the pruned subtree.